File: android/app/src/main/kotlin/com/goalmaker/app/domain/planning/ReviewPrompt.kt
package com.goalmaker.app.domain.planning

/**
 * One prompt from the library (contracts/content/prompts.json): its [id] as reviews store it, the
 * [category] it rotates in, the review kinds it suits, its [text] with `{period}` and `{subject}`
 * still in it, and the [trigger] that makes it worth asking, if any.
 */
data class ReviewPrompt(
    val id: String,
    val category: String,
    val reviews: Set<String>,
    val text: String,
    val trigger: String? = null,
)

File: windows/src/GoalMaker.Core/Planning/ReviewPrompt.cs
namespace GoalMaker.Core.Planning;

/// <summary>
/// One prompt from the library (contracts/content/prompts.json): its <see cref="Id"/> as reviews store
/// it, the <see cref="Category"/> it rotates in, the review kinds it suits, its <see cref="Text"/> with
/// <c>{period}</c> and <c>{subject}</c> still in it, and the <see cref="Trigger"/> that makes it worth
/// asking, if any.
/// </summary>
public sealed record ReviewPrompt(string Id, string Category, IReadOnlySet<string> Reviews, string Text, string? Trigger = null);
